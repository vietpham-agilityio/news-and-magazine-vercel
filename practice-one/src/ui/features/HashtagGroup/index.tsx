// assets
import HashtagFood from "@/images/hashtag/hashtag-food.webp";
import HashtagCar from "@/images/hashtag/hashtag-car.webp";
import HashtagMusic from "@/images/hashtag/hashtag-music.webp";
import HashtagSport from "@/images/hashtag/hashtag-sport.webp";
import HashtagAnimal from "@/images/hashtag/hashtag-animal.webp";
import HashtagAbstract from "@/images/hashtag/hashtag-abstract.webp";

// components
import Image, { StaticImageData } from "next/image";

interface IHashtag {
  id: string;
  name: string;
  imageUrl: string | StaticImageData;
}

interface IProps {
  listHashtags?: IHashtag[];
}

const listHashtagsItem: IHashtag[] = [
  {
    id: "1",
    name: "Hashtag Food",
    imageUrl: HashtagFood,
  },
  {
    id: "2",
    name: "Hashtag Animal",
    imageUrl: HashtagAnimal,
  },
  {
    id: "3",
    name: "Hashtag Car",
    imageUrl: HashtagCar,
  },
  {
    id: "4",
    name: "Hashtag Sport",
    imageUrl: HashtagSport,
  },
  {
    id: "5",
    name: "Hashtag Music",
    imageUrl: HashtagMusic,
  },
  {
    id: "6",
    name: "Hashtag Food",
    imageUrl: HashtagFood,
  },
  {
    id: "7",
    name: "Hashtag Abstract",
    imageUrl: HashtagAbstract,
  },
  {
    id: "8",
    name: "Hashtag Animal",
    imageUrl: HashtagAnimal,
  },
];

export const HashtagGroup = ({ listHashtags = listHashtagsItem }: IProps) =>
  listHashtags.map((hashtag) => {
    const { id, name, imageUrl } = hashtag;
    return (
      <div key={id} className="mx-3 py-2.5 rounded-xl">
        <Image src={imageUrl} width={170} height={48} alt={name} />
      </div>
    );
  });
