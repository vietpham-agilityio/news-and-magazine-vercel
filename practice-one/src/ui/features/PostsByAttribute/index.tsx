import { memo } from 'react';

// service
import { getPostDataByAttribute } from '@/services';

// component
import CardPost from '@/ui/components/Card/CardPost';

// type
import { PostVariant, PostResponse } from '@/types';
import { CardType } from '@/types';

export default memo(async function PostByAttribute({
  attribute,
  cardType,
}: {
  attribute: PostVariant;
  cardType: CardType;
}) {
  const { data: postDataResponse } = await getPostDataByAttribute(attribute);
  const typeCardPost = cardType === 'isVertical';

  return (
    <div className="card-group flex gap-6 overflow-x-scroll">
      {postDataResponse.map((post: PostResponse) => {
        const { title, content, imageUrl, authorId, publicationDate } =
          post.attributes;
        return (
          <CardPost
            isVertical={typeCardPost}
            key={post.id}
            id={post.id.toString()}
            title={title}
            alt={title}
            content={content}
            imageSrc={imageUrl}
            authorId={authorId}
            information={publicationDate}
          />
        );
      })}
    </div>
  );
});
