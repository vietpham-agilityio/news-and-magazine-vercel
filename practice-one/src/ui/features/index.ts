export * from '@/ui/features/TagGroup';
export * from '@/ui/features/SidebarCard';
export * from '@/ui/features/RightSidebar';
export * from '@/ui/features/HashtagGroup';

export { default as PostsGrid } from '@/ui/features/PostsGrid';
export { default as PostDetail } from '@/ui/features/PostDetail';
export { default as PostsByAttribute } from '@/ui/features/PostsByAttribute';
