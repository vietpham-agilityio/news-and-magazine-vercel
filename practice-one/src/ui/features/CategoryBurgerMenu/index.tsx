// services
import { getCategoryData } from '@/services';
import { CategoryResponse, IMenu } from '@/types';

// componens
import { BurgerMenu } from '@/ui/components';

export default async function CategoryBurgerMenu() {
  const { data: categoryDataResponse } = await getCategoryData();

  const listCategories: IMenu[] = [];

  categoryDataResponse.map((category: CategoryResponse) => {
    const { id } = category;
    const { name } = category.attributes;

    listCategories.push({
      id: id.toString(),
      name: name,
    });
  });

  return <BurgerMenu listCategories={listCategories} />;
}
