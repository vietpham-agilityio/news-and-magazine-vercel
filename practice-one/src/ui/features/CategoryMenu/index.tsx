// services
import { getCategoryData } from '@/services';
import { CategoryItem, IMenu } from '@/types';

// componens
import { Menu } from '@/ui/components';

export default async function CategoryMenu() {
  const { data: categoryDataResponse = [] } = await getCategoryData();

  const listCategories: IMenu[] = [];

  categoryDataResponse.map((category: CategoryItem) => {
    const { id } = category;
    const { name } = category.attributes;

    listCategories.push({
      id: id.toString(),
      name: name,
    });
  });

  return <Menu name="Categories" listMenu={listCategories} />;
}
