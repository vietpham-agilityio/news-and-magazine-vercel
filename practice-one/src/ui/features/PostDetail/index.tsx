// services
import { getPostDataById } from '@/services';

// componens
import Image from 'next/image';
import { Typography, Tag } from '@/ui/components';

// type
import { FontWeight, Size } from '@/types';

// icons
import { ChatBox, File, Schedule } from '@/ui/components/Icons';

export default async function PostDetail({ postId }: { postId: number }) {
  const postResponse = await getPostDataById(postId);

  const { title, content, imageUrl } = postResponse.data.attributes;

  return (
    <section className="flex flex-1">
      <article className="flex-1 mb-17.5">
        <div className="mb-7.5 lg:bg-white-90 lg:p-7 xl:mb-[130px] xl:h-[604px] rounded-xl">
          <Typography
            tag="h1"
            additionalClasses="text-md font-medium sm:text-xxl sm:font-normal mb-7"
          >
            {title}
          </Typography>
          <div className="relative mx-auto w-[358px] h-[198px] sm:w-full sm:h-[336px] lg:h-[380px] xl:w-full xl:h-[538px] 2xl:h-[598px] 2xl:w-[1072px]">
            <Image
              fill
              objectFit='cover'
              className="rounded-xl"
              src={imageUrl}
              alt={title}
            />
          </div>
        </div>
        <div className="flex w-full justify-evenly mx-auto mb-12 xl:w-2/4">
          <Tag icon={<Schedule />} content="July 14 , 2022" />
          <Tag
            icon={<ChatBox />}
            content="Comments: 35"
            additionalClass="hidden sm:flex"
          />
          <Tag
            icon={<File />}
            content="Categories: Sport"
            additionalClass="hidden sm:flex"
          />
          <Tag icon={<ChatBox />} content="35" additionalClass="sm:hidden" />
          <Tag icon={<File />} content="Sport" additionalClass="sm:hidden" />
        </div>
        <section>
          {/* content */}
          {content?.split('\n').map((line, index) => (
            <Typography
              key={index}
              tag="h2"
              textSize={Size.MD}
              weight={FontWeight.Normal}
              additionalClasses="text-dark-100"
            >
              {line}
            </Typography>
          ))}
        </section>
      </article>
    </section>
  );
}
