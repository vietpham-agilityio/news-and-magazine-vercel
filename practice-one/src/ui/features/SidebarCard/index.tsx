// static image
import ImageStore from "@/constants/images";

// components
import { Title, PaperPost } from "@/ui/components";

// types
import { IPostItem } from "@/types";

interface IProps {
  listPosts?: IPostItem[];
}

const listPostItems: IPostItem[] = [
  {
    id: "1",
    title: "How to Spend the Perfect Day on Croatia’s Most Magical Island",
    content: "Subhead",
    imageUrl: ImageStore.CarImage,
    alt: "Top post",
  },
  {
    id: "2",
    title: "How to Spend the Perfect Day on Croatia’s Most Magical Island",
    content: "Subhead",
    imageUrl: ImageStore.FrogImage,
    alt: "Top post",
  },
  {
    id: "3",
    title: "How to Spend the Perfect Day on Croatia’s Most Magical Island",
    content: "Subhead",
    imageUrl: ImageStore.PersonOne,
    alt: "Top post",
  },
  {
    id: "4",
    title: "How to Spend the Perfect Day on Croatia’s Most Magical Island",
    content: "Subhead",
    imageUrl: ImageStore.FoodImage,
    alt: "Top post",
  },
  {
    id: "5",
    title: "How to Spend the Perfect Day on Croatia’s Most Magical Island",
    content: "Subhead",
    imageUrl: ImageStore.PersonTwo,
    alt: "Top post",
  },
];

export const SidebarCard = ({ listPosts = listPostItems }: IProps) => (
  <div className="px-[15px] py-5 bg-white-90 rounded-xl">
    <Title title="Top Posts" />
    <div className="mt-5">
      {listPosts.map((post, index) => {
        const { id, content, title, imageUrl, alt } = post;

        return (
          <div
            key={id}
            className={`${index !== listPosts.length - 1 && "mb-5"}`}
          >
            <PaperPost
              title={title}
              content={content}
              imageSrc={imageUrl}
              alt={alt}
            />
          </div>
        );
      })}
    </div>
  </div>
);
