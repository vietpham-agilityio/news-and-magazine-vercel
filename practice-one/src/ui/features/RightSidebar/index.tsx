// asset
import AdvertisingOne from '@/images/advertising/advertising-one.png';
import AdvertisingTwo from '@/images/advertising/advertising-two.png';
import AvatarAuthor from '@/images/avatar/avatar-author.jpg';

// componens
import Image from 'next/image';
import { Avatar, Button, Information } from '@/ui/components';
import { TagGroup, SidebarCard } from '@/ui/features';

// icons
import { ChatBox, PaperPlane, Plus, SaveTagSmall } from '@/ui/components/Icons';

export const RightSidebar = () => (
  <aside className="bg-white-100 lg:ml-[23px] lg:w-[360px]">
    {/* tag and author */}
    <div className="flex flex-col-reverse sm:gap-5 sm:flex-row lg:flex-col lg:gap-0">
      <div className="flex flex-col w-[360px]">
        <div className="flex justify-between bg-white-100 mb-[25px]">
          <Button
            isMarginBottom
            label="Share"
            textColor="text-dark-75"
            bgColor="bg-white-90"
            leftIcon={<PaperPlane color="#404040" />}
          />

          <Button
            isMarginBottom
            label="Marking"
            textColor="text-dark-75"
            bgColor="bg-white-90"
            leftIcon={<SaveTagSmall color="#404040" />}
          />
          <Button
            isMarginBottom
            label="Comment"
            textColor="text-dark-75"
            bgColor="bg-white-90"
            leftIcon={<ChatBox color="#404040" />}
          />
        </div>
        <div className="bg-white-90 flex p-[15px] mb-[25px] rounded-xl">
          <Avatar
            width={87}
            height={87}
            url={AvatarAuthor}
            alt="Avatar author"
          />
          <div className="flex-1 ml-2.5 mt-3.5">
            <Information name="Louis Hoebregts" information="27 posts" />
            <div className="mt-2">
              <Button
                label="Follow"
                leftIcon={<Plus />}
                bgColor="bg-primary-90"
                textColor="text-white-100"
              />
            </div>
          </div>
        </div>
      </div>
      <div className="flex-1 mb-[25px]">
        <TagGroup />
      </div>
    </div>
    <div className="mb-[25px] hidden lg:block">
      <SidebarCard />
    </div>
    <div className="flex justify-between gap-5 lg:flex-col">
      <div className="">
        <Image
          width={360}
          height={180}
          src={AdvertisingTwo}
          alt="Advertising"
        />
      </div>
      <div className="hidden sm:block">
        <Image
          width={360}
          height={180}
          src={AdvertisingOne}
          alt="Advertising"
        />
      </div>
    </div>
  </aside>
);
