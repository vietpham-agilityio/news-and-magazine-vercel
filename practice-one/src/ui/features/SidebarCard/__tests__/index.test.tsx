import { render } from '@testing-library/react';

// component
import { SidebarCard } from '@/ui/features';

// mock data
import { listPostItems } from '@/mocks';

let renderSidebarCard: any;

describe('SidebarCard feature component', () => {
  beforeEach(() => {
    renderSidebarCard = render(
      <SidebarCard listPosts={listPostItems} />
    );
  });

  afterEach(() => {
    renderSidebarCard.unmount();
  });

  it('SidebarCard should render macth snapshot', () => {
    expect(renderSidebarCard.asFragment()).toMatchSnapshot();
  });
});
