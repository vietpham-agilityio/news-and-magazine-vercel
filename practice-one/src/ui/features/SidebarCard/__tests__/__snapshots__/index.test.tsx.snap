// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarCard feature component SidebarCard should render macth snapshot 1`] = `
<DocumentFragment>
  <div
    class="px-[15px] py-5 bg-white-90 rounded-xl"
  >
    <div
      class="flex items-center"
    >
      <div
        class="w-1 h-[11px] bg-primary-100 rounded-2xl mr-1.5"
      />
      <h2
        aria-level="2"
        class="text-md font-medium text-dark-100 leading-6"
        role="heading"
      >
        Top Posts
      </h2>
    </div>
    <div
      class="mt-5"
    >
      <div
        class="mb-5"
      >
        <div
          class="flex cursor-pointer undefined"
        >
          <div
            style="width: 87px; height: 87px; position: relative;"
          >
            <img
              alt="Top post"
              class="rounded-xl"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
            />
          </div>
          <div
            class="flex flex-col gap-1.5 ml-2.5 flex-1  my-3"
          >
            <h2
              aria-level="2"
              class="text-xs font-medium text-dark-100 text-ellipsis overflow-hidden undefined"
              role="heading"
            >
              How to Spend the Perfect Day on Croatia’s Most Magical Island
            </h2>
            <h3
              aria-level="3"
              class="text-md font-light 
          text-dark-75 text-ellipsis overflow-hidden
           text-p
           undefined"
              role="heading"
            >
              Subhead
            </h3>
          </div>
        </div>
      </div>
      <div
        class="mb-5"
      >
        <div
          class="flex cursor-pointer undefined"
        >
          <div
            style="width: 87px; height: 87px; position: relative;"
          >
            <img
              alt="Top post"
              class="rounded-xl"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
            />
          </div>
          <div
            class="flex flex-col gap-1.5 ml-2.5 flex-1  my-3"
          >
            <h2
              aria-level="2"
              class="text-xs font-medium text-dark-100 text-ellipsis overflow-hidden undefined"
              role="heading"
            >
              How to Spend the Perfect Day on Croatia’s Most Magical Island
            </h2>
            <h3
              aria-level="3"
              class="text-md font-light 
          text-dark-75 text-ellipsis overflow-hidden
           text-p
           undefined"
              role="heading"
            >
              Subhead
            </h3>
          </div>
        </div>
      </div>
      <div
        class="mb-5"
      >
        <div
          class="flex cursor-pointer undefined"
        >
          <div
            style="width: 87px; height: 87px; position: relative;"
          >
            <img
              alt="Top post"
              class="rounded-xl"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
            />
          </div>
          <div
            class="flex flex-col gap-1.5 ml-2.5 flex-1  my-3"
          >
            <h2
              aria-level="2"
              class="text-xs font-medium text-dark-100 text-ellipsis overflow-hidden undefined"
              role="heading"
            >
              How to Spend the Perfect Day on Croatia’s Most Magical Island
            </h2>
            <h3
              aria-level="3"
              class="text-md font-light 
          text-dark-75 text-ellipsis overflow-hidden
           text-p
           undefined"
              role="heading"
            >
              Subhead
            </h3>
          </div>
        </div>
      </div>
      <div
        class="mb-5"
      >
        <div
          class="flex cursor-pointer undefined"
        >
          <div
            style="width: 87px; height: 87px; position: relative;"
          >
            <img
              alt="Top post"
              class="rounded-xl"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
            />
          </div>
          <div
            class="flex flex-col gap-1.5 ml-2.5 flex-1  my-3"
          >
            <h2
              aria-level="2"
              class="text-xs font-medium text-dark-100 text-ellipsis overflow-hidden undefined"
              role="heading"
            >
              How to Spend the Perfect Day on Croatia’s Most Magical Island
            </h2>
            <h3
              aria-level="3"
              class="text-md font-light 
          text-dark-75 text-ellipsis overflow-hidden
           text-p
           undefined"
              role="heading"
            >
              Subhead
            </h3>
          </div>
        </div>
      </div>
      <div
        class="false"
      >
        <div
          class="flex cursor-pointer undefined"
        >
          <div
            style="width: 87px; height: 87px; position: relative;"
          >
            <img
              alt="Top post"
              class="rounded-xl"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
            />
          </div>
          <div
            class="flex flex-col gap-1.5 ml-2.5 flex-1  my-3"
          >
            <h2
              aria-level="2"
              class="text-xs font-medium text-dark-100 text-ellipsis overflow-hidden undefined"
              role="heading"
            >
              How to Spend the Perfect Day on Croatia’s Most Magical Island
            </h2>
            <h3
              aria-level="3"
              class="text-md font-light 
          text-dark-75 text-ellipsis overflow-hidden
           text-p
           undefined"
              role="heading"
            >
              Subhead
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
